django/db/backends/mysql/introspection.py

```diff
@@ -189,6 +190,31 @@ def get_constraints(self, cursor, table_name):
                 constraints[constraint]['unique'] = True
             elif kind.lower() == "unique":
                 constraints[constraint]['unique'] = True
+        # Add check constraints.
+        if self.connection.features.can_introspect_check_constraints:
+            type_query = """
+                SELECT c.constraint_name, c.check_clause
+                FROM information_schema.check_constraints AS c
+                WHERE
+                    c.constraint_schema = DATABASE() AND
+                    c.table_name = %s
+            """
+            cursor.execute(type_query, [table_name])
+            for constraint, check_clause in cursor.fetchall():
+                # Parse columns.
+                columns = OrderedSet()
+                for statement in sqlparse.parse(check_clause):
+                    for token in statement.flatten():
+                        if token.ttype in [sqlparse.tokens.Name, sqlparse.tokens.Literal.String.Single]:
+                            columns.add(token.value[1:-1])
```

Comment:
If you're slicing `[1:-1]` just to remove backticks (`` ` ``) around table names, I think you need to find some other way. While working on https://github.com/django/django/pull/11452, I tried this function to get the `JSON_VALID` constraint for introspection. After some debugging, I found out that it gets sliced into `son_vali`.

Perhaps using ``.replace('`', '')`` is enough, but I'm not really sure...

Reply:
@laymonage Thanks. I think it is better to replace it with ```.strip('`')```.
@felixxm What's your idea? if you accept this, I can create a simple PR for that.

Reply:
I think that the main issue is in checking tokens with the list of columns, stripping ``'`'`` will not solve anything because even with that you'll get `JSON_VALID` in the list of column. We should use the same logic as in the [SQLite introspection](https://github.com/django/django/blob/c7bef16a749ccbe2dbc8ce6b697dc80146584384/django/db/backends/sqlite3/introspection.py#L286-L291).

Reply:
@felixxm I will check it out. will try to improve it.